File: tests/fixtures/code/java/Sample.java
// SPDX-FileCopyrightText: 2026 The Nikasha Authors
// SPDX-License-Identifier: Apache-2.0
package org.example.parse;

import java.util.ArrayList;
import java.util.List;

public class Sample {
    private final List<String> items;

    public Sample() {
        this.items = new ArrayList<>();
    }

    public void add(String s) {
        items.add(normalize(s));
    }

    private static String normalize(String s) {
        return s.trim().toLowerCase();
    }

    interface Visitor {
        void visit(String item);
    }

    enum Kind {
        A,
        B;

        boolean isA() {
            return this == A;
        }
    }

    static class Walker implements Visitor {
        @Override
        public void visit(String item) {
            System.out.println(item);
        }
    }
}

File: tests/fixtures/code/rust/sample.rs
// SPDX-FileCopyrightText: 2026 The Nikasha Authors
// SPDX-License-Identifier: Apache-2.0
use std::collections::HashMap;

macro_rules! bail {
    ($msg:expr) => {
        return Err(make_error($msg))
    };
}

pub struct Cache<K> {
    map: HashMap<K, u32>,
}

pub enum Mode {
    Fast,
    Safe,
}

pub trait Store {
    fn get(&self, k: &str) -> Option<u32>;
    fn has(&self, k: &str) -> bool {
        self.get(k).is_some()
    }
}

impl<K: std::hash::Hash + Eq> Cache<K> {
    pub fn new() -> Self {
        Cache { map: HashMap::new() }
    }

    fn put(&mut self, k: K, v: u32) {
        self.map.insert(k, v);
    }
}

fn make_error(msg: &str) -> String {
    msg.to_string()
}

pub fn run(input: &str) -> Result<u32, String> {
    if input.is_empty() {
        bail!("empty");
    }
    println!("{}", checksum(input));
    let n = parse_num(input)?;
    Ok(n)
}

mod util {
    pub fn helper() -> u32 {
        super::make_error("x").len() as u32
    }
}
